gpt_explainer/gpt_slide_expander.py

```diff
@@ -0,0 +1,272 @@
+        """
+        if slide_text and slide_text != ' ':
+            messages = [
+                {
+                    "role": "system",
+                    "content": "You are a chatbot that generates expanded explanations about slide topics."
+                },
+                {
+                    "role": "user",
+                    "content": f"Please provide an expanded explanation for the slide content. "
+                               f"The slide topic is: {slide_text}. "
+                               f"If applicable, provide the explanation within the context of the "
+                               f"main topic: {presentation_topic}. "
+                               f"Please make the explanation natural and expand on the slide topic with"
+                               f" relevant details. make the explanation up tp 3 sentences"
+                }
+            ]
+
+            gpt_response = openai.ChatCompletion.create(
+                model="gpt-3.5-turbo",
+                messages=messages
+            )
+
+            return gpt_response['choices'][0]['message']['content']
+        else:
+            return 'No text in the slide - unable to generate explanation'
+
+    @staticmethod
+    async def process_individual_slide_explanation_safely(slide_text: str, presentation_topic: str, max_retry: int,
+                                                          slide_index: int) -> str:
+        """
+        Process an individual slide task for generating an explanation.
+        Args:
+            slide_text: The text of the slide to generate an explanation for.
+            presentation_topic: The topic or context of the presentation.
+            max_retry: The maximum number of retries for generating the explanation.
+            slide_index: The index of the slide being processed.
+        Returns:
+            The generated explanation for the slide, or an error message if an `openai.error` occurs.
+        Raises:
+            openai.error: If an OpenAI API error occurs during explanation generation for the slide.
+        """
+        try:
+            return await GptSlideExpander.generate_explanation_for_slide_with_retry(slide_text, presentation_topic,
+                                                                                    max_retry)
+        except openai.error as e:
+            return f"ERROR - explanation generation for slide {slide_index} failed: {str(e)}"
+
+    def run_explainer(self) -> None:
+        """
+        Runs the GPT explainer system indefinitely, processing files dropped into the 'uploads' folder.
+        The explainer sleeps for 10 seconds between iterations.
+        """
+        while True:
+            files_to_process = self.get_files_to_process()
+            for file_path in files_to_process:
+                self.process_file(file_path)
+                time.sleep(10)  # Sleep for 10 seconds between processing each file
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.